patches/net/minecraft/world/level/chunk/ChunkAccess.java.patch

```diff
@@ -9,35 +9,35 @@
      public static final int NO_FILLED_SECTION = -1;
      private static final Logger LOGGER = LogUtils.getLogger();
      private static final LongSet EMPTY_REFERENCE_SET = new LongOpenHashSet();
-@@ -308,23 +_,28 @@
+@@ -308,10 +_,19 @@
  
      @Override
      public final void findBlockLightSources(BiConsumer<BlockPos, BlockState> p_285269_) {
 -        this.findBlocks(p_284897_ -> p_284897_.getLightEmission() != 0, p_285269_);
-+        this.findBlocks((p_284897_, pos) -> p_284897_.getLightEmission(this, pos) != 0, p_285269_);
++        this.findBlocks(p_284897_ -> p_284897_.hasDynamicLightEmission() || p_284897_.getLightEmission(net.minecraft.world.level.EmptyBlockGetter.INSTANCE, BlockPos.ZERO) != 0, (p_284897_, pos) -> p_284897_.getLightEmission(this, pos) != 0, p_285269_);
```

Comment:
(This comment does not need to block merging, feel free to resolve it if necessary.)

In 20.5, do we plan to stop calling the positional `getLightEmission` here?

Reply:
I'm a bit torn on this. On the one hand, going back to the non-positional vanilla `getLightEmission()` method would completely cut out any potential of calling into a dynamic light getter implementation, which would be nice. On the other hand, I can see people overriding the positional method for purely state-based light instead of specifying that as part of the block's properties (the light emission is stored on the `BlockState` and as such there's no overridable method in `Block` for purely state-based light emission). I think this needs more discussion.